### DISCUSSION TITLE
Question about socket.io

### DISCUSSION THREAD

**Discussion by paul-bell** [enisdenjo/graphql-ws#378] [Q&A] [answered]
Hello,

Your npmjs.com page shows Web Socket servers realized via:

- ws
- uWebSockets.js
- fastify-websocket

Instead of these, can I use [socket.io](https://socket.io/) together with graphql-ws?

Thank you.

**Comment by enisdenjo** [comment, marked as answer]
`graphql-ws` and it's accompanying [GraphQL over WebSocket Protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) is purely for WebSockets. Since Socket.io leverages more than just WebSockets (HTTP long-polling), `graphql-ws` will not be supporting it.

Still, you should be able to implement the server on top of Socket.io using [`makeServer`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/modules/server.md#makeserver) (see ["Server usage with ws" recipe](https://github.com/enisdenjo/graphql-ws#ws) for a quick start). **However,** you'll have to implement the client from scratch completely on your own.

**Comment by n1ru4l** [comment]
Hey, saw this issue and wanted to share that a while back I built a GraphQL over Socket.io transport:

- https://github.com/n1ru4l/graphql-live-query/tree/main/packages/socket-io-graphql-server
- https://github.com/n1ru4l/graphql-live-query/tree/main/packages/socket-io-graphql-client

**Disclaimer** that it does not follow the GraphQL over WebSocket Protocol at all. It basically only glues together the graphql-js functionality with Socket.io. I never used it in a large-scale deployment and only for an app that you host for ~15 concurrent users.
